File: backup.js
const fs = require('fs');
const path = require('path');
const moment = require('moment');

const backupDir = path.join(__dirname, 'Backup');
const projectDir = __dirname;

if (!fs.existsSync(backupDir)){
    fs.mkdirSync(backupDir);
}

function backup() {
    const timestamp = moment().format('YYYYMMDDHHmmss');
    const backupPath = path.join(backupDir, `backup_${timestamp}.zip`);

    const exec = require('child_process').exec;

    exec(`zip -r ${backupPath} ${projectDir} -x ${backupDir}/*`, (error) => {
        if (error) {
            console.error(`Error creating backup: ${error}`);
        } else {
            console.log(`Backup created at ${backupPath}`);
        }
    });
}

setInterval(backup, 3 * 60 * 1000); // 3 minutes

File: backup.py
import shutil
import os
import time
from datetime import datetime

backup_folder = 'Backup'
interval = 180  # интервал в секундах (180 секунд = 3 минуты)

if not os.path.exists(backup_folder):
    os.makedirs(backup_folder)

while True:
    timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
    backup_name = f'backup_{timestamp}.zip'
    backup_path = os.path.join(backup_folder, backup_name)
    
    shutil.make_archive(backup_path.replace('.zip', ''), 'zip', '.')

    print(f'Backup created: {backup_path}')
    
    time.sleep(interval)
